### PR TITLE
Fix inline protocol can't parse LF only EOL

### LINKED ISSUE #807: redis-cli --pipe syntax error
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Version

kvrocks unstable 40989fc26dc4229bd9f797c14eaa4df07574f47e
ubuntu 22.04

### Minimal reproduce step

```
root@localhost:~# cat /tmp/text1.txt
set zzzap zzzzzzzzzzzzzzzzznologin
set vv zzzzzzzzzzzzzzzzpsbinnologin
```

```
root@localhost:~# cat /tmp/text1.txt | redis-cli --pipe --verbose
```

### What did you expect to see?

This is with redis or keydb 

```
root@localhost:~# cat /tmp/text1.txt | redis-cli -h xx.79.179.186 --pipe --verbose
All data transferred. Waiting for the last reply...
Last reply received from server.
errors: 0, replies: 2
```

### What did you see instead?

root@localhost:~# cat /tmp/text1.txt | redis-cli --pipe --verbose
All data transferred. Waiting for the last reply...
ERR syntax error
Last reply received from server.
errors: 1, replies: 1

### Anything Else?

> redis-cli --version
redis-cli 6.0.16

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #807

### CODE REVIEW

**Review by tisonkun — COMMENTED** [review]
In the [RESP spec](https://redis.io/docs/reference/protocol-spec/) it seems only CRLF("\r\n") is the allowed delimiter. Does Redis implement a different manner actually?

**Comment by git-hulk** [comment]
> In the [RESP spec](https://redis.io/docs/reference/protocol-spec/) it seems only CRLF("\r\n") is the allowed delimiter. Does Redis implement a different manner actually?

Yes, RESP only allows CRLF but inline mode allows the LF end of line as well: https://github.com/redis/redis/blob/unstable/src/networking.c#L2080

File: src/redis_request.cc [hunk 1]
```diff
@@ -61,13 +71,13 @@ Status Request::Tokenize(evbuffer *input) {
           } catch (std::exception &e) {
             return Status(Status::NotOK, "Protocol error: invalid multibulk length");
           }
+          if (isOnlyLF || multi_bulk_len_ > (int64_t)PROTO_MULTI_MAX_SIZE) {
+            return Status(Status::NotOK, "Protocol error: invalid multibulk length");
+          }
```
Comment (tisonkun) [inline src/redis_request.cc:76]: Why do we disallow `\n` as EOL in the multi-bulk scenario?

**Review by tisonkun — APPROVED** [review]

**Review by tisonkun — COMMENTED** [review]
~~I try `echo -e 'PING' | nc localhost 6666` with empty reply while `echo -e 'PING' | nc localhost 6379` with `PONG`. Shall we handle no eol delimiter at the end case?~~

Sorry, I get it wrong. `echo -e` automatically appends a `\n`

File: src/redis_request.cc (hunk 1 above)
Comment (git-hulk) [reply src/redis_request.cc:76]: Redis also disallowed \n in multi bulk protocol, so we just keep consistent with Redis here.

**Review by PragmaTwice — APPROVED** [review]

File: src/redis_request.cc (hunk 1 above)
Comment (tisonkun) [reply src/redis_request.cc:76]: Thanks for your explanation. I get the point now.

**Pull request merged**